www/js/diary/current.js

```diff
@@ -37,39 +79,60 @@
     hours = hours % 12;
     hours = hours ? hours : 12;
     minutes = minutes < 10 ? '0'+ minutes : minutes;
-    $scope.startTime = hours + ':' + minutes + ' ' + amOrPm;
+    return hours + ':' + minutes + ' ' + amOrPm;
+  };
+
+  var deg2rad = function(deg) {
+    return deg * (Math.PI/180);
+  };
+
+  // Haversine formula
+  var getMeterfromLglat = function(curr_lglat, last_lglat) {
```

Comment:
you can use Leaflet `distanceTo` instead of re-implementing. I think that is what I use.